src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -664,8 +659,7 @@ uct_dc_mlx5_ep_flush_cancel(uct_dc_mlx5_ep_t *ep)
 
 static uint32_t uct_dc_mlx5_flush_remote_rkey(uct_dc_mlx5_ep_t *ep)
 {
-    return ((uint32_t)ep->flush_remote_rkey << 16) +
-           ((uint32_t)ep->atomic_mr_id << 8);
+    return ((uint32_t)ep->flush_remote_rkey << 16) + ep->atomic_mr_offset;
```

Comment:
maybe assert that lowest bits are `0`  in `ep->atomic_mr_offset`.

Reply:
yep, good idea - added assert

Reply:
1. document
2. maybe use | instead of +